src/FSLibrary/StellarCoreHTTP.fs

```diff
@@ -764,6 +764,19 @@ type Peer with
 
         if self.IsLoadGenComplete() <> Success then failwith "Loadgen failed!"
 
+    member self.WaitForLoadGenCompleteWithAllowedFailures (loadGen: LoadGen) (allowedFailureRate: float) =
+        RetryUntilTrue
+            (fun _ -> (self.IsLoadGenComplete() = Success || self.IsLoadGenComplete() = Failure))
+            (fun _ ->
+                self.LogLoadGenProgressTowards loadGen
+                self.LogCoreSetListStatusWithTiers self.networkCfg.CoreSetList)
+
+        let m : Metrics.Metrics = self.GetMetrics()
+        let txBanRate : float = (float m.HerderPendingTxsBanned.Value.Count) / (float loadGen.txs)
```

Comment:
I'm not sure if this is the correct metric to check here, but we can't check metrics about TXs succeeding because we don't care if TXs actually succeed or fail. Maybe we should use the sum of `ledger.apply.success` and `ledger.apply.failure` here instead so we get a count of all TXs that actually made it through to apply phase?